cli/src/config.rs

```diff
@@ -0,0 +1,143 @@
+use color_eyre::Result;
+use hyper::{http::HeaderName, Method};
+use serde::Deserialize;
+use serde_with::{serde_as, DisplayFromStr};
+use std::{fs::File, io::BufReader, net::SocketAddr, path::PathBuf, time::Duration};
+use tower_http::cors::CorsLayer;
+
+pub type Timeout = u64;
+
+/// Server Configuration passed on initialization
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    /// Server address <host:port>
+    pub address: SocketAddr,
+
+    /// Cors configuration
+    pub cors: Cors,
+
+    /// Max Time to Responce, in milliseconds
+    pub request_timeout: Timeout,
+
+    /// [Stub] File to save logs
+    pub log_file: Option<PathBuf>,
+
+    /// Tracing Level
+    #[serde(default = "tracing_default")]
+    #[serde_as(as = "DisplayFromStr")]
+    pub trace_level: tracing::Level,
+}
+
+const fn tracing_default() -> tracing::Level {
+    tracing::Level::INFO
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            address: SocketAddr::from(([0, 0, 0, 0], 6000)),
+            log_file: None,
+            trace_level: tracing::Level::INFO,
+            cors: Cors::default(),
+            request_timeout: 5000,
+        }
+    }
+}
+
+impl Config {
+    /// Parses the file spcified in `path`
+    ///
+    /// # Errors
+    ///
+    /// The fucntion will fail if either it couldn't open config file
+    /// or failed to parse given file
+    pub fn from_file(path: PathBuf) -> Result<Self> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+        Ok((serde_yaml::from_reader(reader) as Result<Self, _>)?)
+    }
+}
+
+/// Cors Configuration struct
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Cors {
```

Comment:
This is an overly detailed configuration. And in the production it will never change. From this point of view, it should be removed from the config, or reduced to a single parameter `CorsAllowAll: bool` with default value =`false`